core/lexer.cpp

```diff
@@ -507,6 +507,38 @@ Tokens jsonnet_lex(const std::string &filename, const char *input)
             }
             break;
 
+            // Verbatim string literals.
+            case '@': {
+                c++;
+                if (*c != '"' && *c != '\'') {
+                    std::stringstream ss;
+                    ss << "Couldn't lex verbatim string, junk after '@': " << *c;
+                    throw StaticError(filename, begin, ss.str());
+                }
+                const char quot = *c;
+                c++;  // Advance beyond the opening quote.
+                for (; ; ++c) {
+                    if (*c == '\0')  {
+                        throw StaticError(filename, begin, "Unterminated verbatim string");
+                    }
+                    if (*c == quot) {
+                        if (*(c+1) == quot) {
```

Comment:
Can you add a comment explaining that the '' and "" quoting is interpreted here, unlike the non-verbatim strings where it is done later by jsonnet_string_unescape.  This is OK in this case because no information is lost by resolving the repeated quote into a single quote, so we can go back to the original form in the formatter.